lib/unexpected-knex.js

```diff
@@ -226,6 +226,42 @@ function init(expect) {
     }
   );
 
+  const ascendingOrder = (a, b) => parseInt(a.id) - parseInt(b.id);
+
+  expect.addAssertion(
+    '<knexQuery> to have sorted rows [exhaustively] satisfying <array>',
+    (expect, subject, value) => {
+      const exhaustive = expect.flags.exhaustively;
+      return expect(
+        subject,
+        'to have rows satisfying',
+        expect.it(
+          'sorted by',
+          ascendingOrder,
+          exhaustive ? 'to exhaustively satisfy' : 'to satisfy',
+          value
+        )
+      );
+    }
+  );
+
+  expect.addAssertion(
+    '<Promise> to be fulfilled with sorted rows [exhaustively] satisfying <array>',
```

Comment:
For non-overloaded assertions like this one the likelihood of a collision is very low, so I think it's fine. If you could add a `KnexPromise` type that identifies the specific return values from your library it would be slightly better, but maybe not worth it.

If you're only using this assertion as a helper within your plugin, you also have the option to use [`expect.child`](https://unexpected.js.org/api/child/) within your plugin and only `exportAssertion`/`exportType` the things the user actually needs to have available. Then internal assertions can be added with `addAssertion` on the child expect so they don't leak out.